internal/util/os/os.go

```diff
@@ -10,7 +10,10 @@ import (
 // Getenv returns an optional String for the environment variable with given
 // key.
 func GetOptEnv(key string) opt.String {
-	return opt.NewNotEmptyString(os.Getenv(key))
```

Comment:
Isn't this doing the same thing? `Getenv` will return `""` if the key is missing, which will be converted to `opt.NullString` by `NewNotEmptyString()`.

Reply:
Unfortunately no, after a few tests I ended up using the recommendation on the [docs](https://pkg.go.dev/os#Getenv).

> Getenv retrieves the value of the environment variable named by the key. It returns the value, which will be empty if the variable is not present. ***To distinguish between an empty value and an unset value, use LookupEnv.***

So if you call `Getenv` with an unset environment variable the value returned from that function is not `""`.

I found the `LookupEnv` in other parts of the code, so I figured you had already bumped into this.